### PR TITLE
[mlir][nfc] Update tests for Contract -> Op transforms

### PR DESCRIPTION
**Pull request by banach-space**
Updates two tests for vector.contract -> vector.outerproduct
transformations:

1. Rename "vector-contract-to-outerproduct-transforms.mlir" as
   "vector-contract-to-outerproduct-matmul-transforms.mlir". The new
   name more accurate captures what's being tested. it is also
   consistent with
   "vector-contract-to-outerproduct-matvec-transforms.mlir", which
   covers vector matvec operations and makes finding relevant tests
   easier.

2. For matmul tests, move the traits definining the iteration spaces to
   the top of the file. This is consistent with how matvec tests are
   defined and also makes it easy to quickly identify what cases are
   covered.

3. For matmul tests, use more meaningful names for function arguments.
   This helps keep things consistent across the file (i.e. function
   definitions wih check lines and comments).

4. For matvec test, move a few tests around so that the most basic case
   (without masking) is first.

5. Update comments.

### CODE REVIEW

**Comment by banach-space** [comment]
@MacDue I've split this one into 5 commits to make it easier to review. Hope this helps.

File: mlir/test/Dialect/Vector/vector-contract-to-outerproduct-matmul-transforms.mlir [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
-// output matrix) should work as well.
-//
-// (*) The conversion tested in this file unrolls along the reduction
-// dimension, which is not supported for scalable vectors.
+/// Tests for `vector.contract` -> `vector.outerproduct` transformations for
+/// Matmul operations:
+///   C += A * B.
+/// (A, B and C are 2-d matrices). ATM three different variants / are tested:
+///   * plain (no mask, fixed-wdith vectors),
+///   * masked (fixed-width vectors,
```
Comment (MacDue) [inline mlir/test/Dialect/Vector/vector-contract-to-outerproduct-matmul-transforms.mlir:8]:
```suggestion
///   * masked (fixed-width vectors),
```

File: mlir/test/Dialect/Vector/vector-contract-to-outerproduct-matmul-transforms.mlir [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
+///   * scalable (mask + scalable vectors).
+/// In order for the "vector.contract -> vector.outerproduct" patterns to work,
+/// only the non-reduction dimension can be scalable (*). For Matmul operations
```
Comment (MacDue) [inline mlir/test/Dialect/Vector/vector-contract-to-outerproduct-matmul-transforms.mlir:11]:
```suggestion
/// only the non-reduction dimension can be scalable (*). For matmul operations
```

**Review by MacDue — APPROVED** [review]
LGTM, thanks for splitting things up! Just some very minor nits:

**Pull request merged**